spec/requests/complaints_spec.rb

```diff
@@ -7,9 +7,26 @@
       expect(response).to have_http_status(200)
     end
 
-    it "renders the text" do
-      get root_path
-      expect(response.body).to include("My complaints")
+    describe "without an authorized user" do
+      it "asks the user to log" do
+        get root_path
+        expect(response.body).to include("Please log in to access your complaints list")
+      end
+    end
+
+    describe "with an authorized user" do
+      before do
+        user = {
+          name: "Test User",
+          email: "testuser@example.com"
+        }.with_indifferent_access
+        allow_any_instance_of(ActionDispatch::Request::Session).to receive(:[]).and_return user
```

Comment:
```suggestion
        allow_any_instance_of(ActionDispatch::Request::Session).to receive(:[]).with("user").and_return user
```

that feels like it might be more future-proof if/when we add other things to the session.